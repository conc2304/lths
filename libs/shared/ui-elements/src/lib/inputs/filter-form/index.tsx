import { Dialog, DialogContent, DialogActions, Divider, useTheme, useMediaQuery, Box } from '@mui/material';
import Button from '@mui/material/Button';
import { FormSchema, FormState } from '@lths/shared/ui-filters';

import { FormTitle } from './form-title';
import { ChipContainer } from '../chip-container';
import { Form } from '../form-builder';

export type FilterFormProps = {
  title: string;
  chipTitle?: string;
  open: boolean;
  filterSchema: FormSchema[];
  handleApplyFilters: (formData: FormState) => void;
  handleClose: () => void;
  handleClearFilters: () => void;
  handleCancel: () => void;
  formState: FormState;
  removeItem: (parentID: string, itemID: string) => void;
};

export const FilterForm = ({
  title = 'Apply Filters',
  chipTitle = 'Selected Filters',
  filterSchema,
  handleClose,
  handleApplyFilters,
  handleClearFilters,
  handleCancel,
  formState,
  removeItem,
  open,
}: FilterFormProps) => {
  const theme = useTheme();
  const fullScreen = useMediaQuery(theme.breakpoints.down('sm'));

  const handleOnApplyFilters = () => {
    handleApplyFilters(formState);
    handleClose();
  };

  const handleChipDelete: typeof removeItem = (parentID, itemID) => {
    removeItem && removeItem(parentID, itemID);
  };

  return (
    <Dialog
      open={open}
      aria-labelledby="filter-dialog-title"
      className="FilterForm--Dailog"
      maxWidth="md"
      fullWidth={true}
      fullScreen={fullScreen}
    >
      <Box px="2.675rem" pt="2.125rem">
        <Box>
          <FormTitle id="filter-dialog-title" onClose={handleCancel}>
            {title}
          </FormTitle>
        </Box>
        <DialogContent>
          <Form formSchema={filterSchema} />
          <Box mt={7}>
            <ChipContainer title={chipTitle} onDelete={handleChipDelete} selectedFilters={formState} />
          </Box>
        </DialogContent>
        <Divider variant="middle" sx={{ mt: 3 }} />
        <DialogActions sx={{ m: 3 }}>
          <Button
            variant="text"
            color="secondaryButton"
            onClick={handleClearFilters}
            sx={{ fontSize: '0.75rem', mr: 1.75 }}
          >
            CLEAR ALL FILTERS
          </Button>
          <Button variant="outlined" color="primary" onClick={handleCancel} sx={{ mr: 0.75 }}>
            CANCEL
          </Button>
          <Button variant="contained" color="primary" onClick={handleOnApplyFilters}>
            APPLY FILTERS
          </Button>
        </DialogActions>
      </Box>
    </Dialog>
  );
};
