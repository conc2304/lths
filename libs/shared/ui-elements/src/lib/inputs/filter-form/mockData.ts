import { FormSchema, FormState } from '../../ui-filters/types';

export const formStateMock: FormState = {
  event_state: {
    pre_game: {
      title: 'Pre-game',
      id: 'pre_game',
    },
  },
};

export const formSchemaMock: FormSchema[] = [
  {
    title: 'Event States',
    info: {
      description: 'Mock Description of filter',
      url: 'linkto.info.com',
    },
    id: 'event_state',
    seq: [0, 1],
    type: null,
    data: [
      {
        type: 'checkbox',
        title: null,
        seq: -10,
        default_value: ['pre_game'],
        data: [
          {
            title: 'Pre-game',
            id: 'pre_game',
            seq: 1,
          },
          {
            title: 'No Event',
            id: 'no_event',
            seq: 5,
          },
          {
            title: 'During the Event',
            id: 'during_event',
            seq: 3,
          },
          {
            title: 'Event day',
            id: 'event_day',
            seq: 2,
          },
          {
            title: 'Post event',
            id: 'post_event',
            seq: 4,
          },
        ],
      },
    ],
  },
  {
    title: 'Location',
    id: 'location',
    info: {
      description: 'Mock Description of Location Filter',
      url: 'example.com',
    },
    seq: [1, 0],
    data: [
      {
        type: 'checkbox',
        title: null,
        seq: 2,
        default_value: ['at_home'],
        data: [
          {
            title: 'At arena',
            id: 'at_arena',
            seq: 1,
          },
          {
            title: 'At home',
            id: 'at_home',
            seq: 2,
          },
        ],
      },
    ],
  },
  {
    title: 'User Segments',
    id: 'user_segments',
    type: null,
    info: {
      url: 'url.com',
      description: 'Info About User Segments',
    },
    seq: [3, 10],
    data: [
      {
        type: 'checkbox',
        title: null,
        default_value: [],
        data: [
          {
            title: 'Single-event Attendees',
            id: 'single_event_attendees',
          },
          {
            title: 'Mighty Program Members',
            id: 'mighty_program_members',
          },
          {
            title: 'Premium Members',
            id: 'premium_members',
          },
        ],
      },
    ],
  },
];
