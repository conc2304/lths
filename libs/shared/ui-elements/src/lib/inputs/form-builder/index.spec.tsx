import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';

import { Form } from './index';
import { groupItemsBySeq } from '../../inputs/filter-form/utils';
import { FormSchema, FormState, AddItem, RemoveItem, ClearGroup, AddGroupItems } from '../../ui-filters';
import { formSchemaMock, formStateMock } from '../filter-form/mockData';

describe('Form component', () => {
  const formSchema: FormSchema[] = formSchemaMock;
  const formState: FormState = formStateMock;
  const columnGroups = groupItemsBySeq([...formSchema]);

  const onAddItem: AddItem = jest.fn();
  const onRemoveItem: RemoveItem = jest.fn();
  const onClearGroup: ClearGroup = jest.fn();
  const onAddGroupItems: AddGroupItems = jest.fn();

  it('should render the form with provided schema and state', () => {
    // Arrange
    render(
      <Form
        formSchema={formSchema}
        formState={formState}
        onAddItem={onAddItem}
        onRemoveItem={onRemoveItem}
        onClearGroup={onClearGroup}
        onAddGroupItems={onAddGroupItems}
      />
    );

    // Act

    // Assert
    for (const item of formSchema) {
      const title = item.title;
      if (!title) return;
      expect(screen.getByText(title)).toBeInTheDocument();
    }
  });

  it('should render the correct number of columns based on the formSchema', () => {
    render(
      <Form
        formSchema={formSchema}
        formState={formState}
        onAddItem={onAddItem}
        onRemoveItem={onRemoveItem}
        onClearGroup={onClearGroup}
        onAddGroupItems={onAddGroupItems}
      />
    );

    const columns = screen.getAllByTestId('form-column');
    expect(columns.length).toBe(Object.keys(columnGroups).length);
  });

  it('should render the correct number of VerticalFormGroup components based on the formSchema', () => {
    // Arrange
    render(
      <Form
        formSchema={formSchema}
        formState={formState}
        onAddItem={onAddItem}
        onRemoveItem={onRemoveItem}
        onClearGroup={onClearGroup}
        onAddGroupItems={onAddGroupItems}
      />
    );

    const formGroupsCount = Object.values(columnGroups).reduce(
      (count, currentColumn) => count + currentColumn.length,
      0
    );
    const verticalFormGroups = screen.getAllByTestId('vertical-form-group');

    // Act
    // Assert
    expect(verticalFormGroups.length).toBe(formGroupsCount);
  });
});
