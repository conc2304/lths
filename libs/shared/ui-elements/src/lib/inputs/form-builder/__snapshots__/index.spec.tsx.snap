// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<body>
  <div>
    <div
      data-testid="form-builder"
    >
      <div
        class="MuiGrid-root MuiGrid-container Form--root css-1oovuve-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item FormColumn--root css-13i4rnv-MuiGrid-root"
          data-testid="form-column"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column FormColumnRowContainer--root css-1r2s251-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item FormRow--root css-13i4rnv-MuiGrid-root"
              data-testid="vertical-form-group"
            >
              <div
                class="VerticalFormGroup-root"
              >
                <div
                  class="VerticalFormGroup--title"
                >
                  <div
                    class="MuiBox-root css-9xkke2"
                  >
                    <h4
                      class="MuiTypography-root MuiTypography-h4 css-64o0nh-MuiTypography-root"
                    >
                      Event States
                    </h4>
                    <div
                      class="MuiBox-root css-h38yef"
                    >
                      <div>
                        <div
                          aria-haspopup="true"
                          class="MuiBox-root css-tzwgne"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18hqis9-MuiSvgIcon-root"
                            data-testid="InfoOutlinedIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiFormGroup-root css-163p3vr-MuiFormGroup-root"
                >
                  <div
                    class="MuiStack-root css-r09xa8-MuiStack-root"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-indeterminate MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-indeterminate MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-indeterminate MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="true"
                          type="checkbox"
                          value="States--togle"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="IndeterminateCheckBoxIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        All States
                      </span>
                    </label>
                    <hr
                      class="MuiDivider-root MuiDivider-middle css-1iy69km-MuiDivider-root"
                    />
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          checked=""
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="pre_game"
                          type="checkbox"
                          value="pre_game"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Pre-game
                      </span>
                    </label>
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="event_day"
                          type="checkbox"
                          value="event_day"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Event day
                      </span>
                    </label>
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="during_event"
                          type="checkbox"
                          value="during_event"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        During the Event
                      </span>
                    </label>
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="post_event"
                          type="checkbox"
                          value="post_event"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Post event
                      </span>
                    </label>
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="no_event"
                          type="checkbox"
                          value="no_event"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        No Event
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item FormColumn--root css-13i4rnv-MuiGrid-root"
          data-testid="form-column"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column FormColumnRowContainer--root css-1r2s251-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item FormRow--root css-13i4rnv-MuiGrid-root"
              data-testid="vertical-form-group"
            >
              <div
                class="VerticalFormGroup-root"
              >
                <div
                  class="VerticalFormGroup--title"
                >
                  <div
                    class="MuiBox-root css-9xkke2"
                  >
                    <h4
                      class="MuiTypography-root MuiTypography-h4 css-64o0nh-MuiTypography-root"
                    >
                      Location
                    </h4>
                    <div
                      class="MuiBox-root css-h38yef"
                    >
                      <div>
                        <div
                          aria-haspopup="true"
                          class="MuiBox-root css-tzwgne"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18hqis9-MuiSvgIcon-root"
                            data-testid="InfoOutlinedIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiFormGroup-root css-163p3vr-MuiFormGroup-root"
                >
                  <div
                    class="MuiStack-root css-r09xa8-MuiStack-root"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          type="checkbox"
                          value="Locations--togle"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        All Locations
                      </span>
                    </label>
                    <hr
                      class="MuiDivider-root MuiDivider-middle css-1iy69km-MuiDivider-root"
                    />
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="at_arena"
                          type="checkbox"
                          value="at_arena"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        At arena
                      </span>
                    </label>
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="at_home"
                          type="checkbox"
                          value="at_home"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        At home
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item FormColumn--root css-13i4rnv-MuiGrid-root"
          data-testid="form-column"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column FormColumnRowContainer--root css-1r2s251-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item FormRow--root css-13i4rnv-MuiGrid-root"
              data-testid="vertical-form-group"
            >
              <div
                class="VerticalFormGroup-root"
              >
                <div
                  class="VerticalFormGroup--title"
                >
                  <div
                    class="MuiBox-root css-9xkke2"
                  >
                    <h4
                      class="MuiTypography-root MuiTypography-h4 css-64o0nh-MuiTypography-root"
                    >
                      User Segments
                    </h4>
                    <div
                      class="MuiBox-root css-h38yef"
                    >
                      <div>
                        <div
                          aria-haspopup="true"
                          class="MuiBox-root css-tzwgne"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18hqis9-MuiSvgIcon-root"
                            data-testid="InfoOutlinedIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiFormGroup-root css-163p3vr-MuiFormGroup-root"
                >
                  <div
                    class="MuiStack-root css-r09xa8-MuiStack-root"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          type="checkbox"
                          value="Segments--togle"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        All Segments
                      </span>
                    </label>
                    <hr
                      class="MuiDivider-root MuiDivider-middle css-1iy69km-MuiDivider-root"
                    />
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="single_event_attendees"
                          type="checkbox"
                          value="single_event_attendees"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Single-event Attendees
                      </span>
                    </label>
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="mighty_program_members"
                          type="checkbox"
                          value="mighty_program_members"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Mighty Program Members
                      </span>
                    </label>
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="premium_members"
                          type="checkbox"
                          value="premium_members"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Premium Members
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
